### ISSUE TITLE
Activity dashboard empty states

### ISSUE THREAD

**Issue by codemonkey800** [chanzuckerberg/napari-hub#732]
Empty states need to be implemented for the activity dashboard and area chart components

<img width="726" alt="image" src="https://user-images.githubusercontent.com/2176050/200068777-79e40c64-f2e9-4437-b0f4-6868e76cd24c.png">

<img width="470" alt="image" src="https://user-images.githubusercontent.com/2176050/200068802-dd500dff-797f-4373-acb5-8620bc208827.png">

**Comment by richaagarwal** [comment]
@codemonkey800 Hmm when I go to a new plugin on staging, e.g. https://staging.napari-hub.org/plugins/napari-psf-analysis, I don't see any activity section. Is this expected, or is there corresponding back-end work that needs to happen as a result of your changes?

**Comment by richaagarwal** [comment]
@codemonkey800 Shouldn't this plugin https://staging.napari-hub.org/plugins/napari-svetlana have the "There is not yet enough data to display for this plugin" for the monthly installs section?

**Comment by richaagarwal** [comment]
Ah I would suggest using the release date to calculate this, but yes it seems back-end and front-end should align here on what data needs to be passed. Let's discuss more on Monday!

**Comment by manasaV3** [comment]
I agree that using the release date is ideal, but we don't store the release date in the activity data we currently cache and fetch on API calls. Utilizing the min on the timestamp of installs will not accurately represent the release date in cases where a plugin didn't have installs in the first few months after release. 
From an earlier discussion, I believe there was a consensus that exposing the release date instead of the total months in the stats would be valuable. This would alleviate the problem but might require a bit of rework in the back end. 

As the front end already has the release date, we could leverage that and add a check before processing to render the timeline. This could be a stop-gap before we implement the release date being returned as a part of the stats. 

This could also be implemented using the total months metric that gets returned right now. But @klai95 and I noticed a bug with this. We are currently using the minimum timestamp of installs to compute the total months, which would not be accurate in the edge case discussed above.

**Comment by richaagarwal** [comment]
> I agree that using the release date is ideal, but we don't store the release date in the activity data we currently cache and fetch on API calls. Utilizing the min on the timestamp of installs will not accurately represent the release date in cases where a plugin didn't have installs in the first few months after release. From an earlier discussion, I believe there was a consensus that exposing the release date instead of the total months in the stats would be valuable. This would alleviate the problem but might require a bit of rework in the back end.
> 
> As the front end already has the release date, we could leverage that and add a check before processing to render the timeline. This could be a stop-gap before we implement the release date being returned as a part of the stats.
> 
> This could also be implemented using the total months metric that gets returned right now. But @klai95 and I noticed a bug with this. We are currently using the minimum timestamp of installs to compute the total months, which would not be accurate in the edge case discussed above.

@manasaV3 Great points - let's move forward with Jeremy's fix for now, but the total months bug sounds like its own separate issue; could you create an issue to track that bug and your proposed fix?

**Comment by klai95** [comment]
> @richaagarwal @klai95 #772 should fix the issue, but I think long term in the future we'll want to trim the amount of data we return based on the release date to save bandwidth when sending data to the frontend. it's not significant right now, but has the potential to impact users on slower networks as we add more data to the metrics endpoint

Just to clarify, essentially we would use the release date as the start date to for filtering data (as in we would discard data before that start date). After the filtering process, in the case (1) there are fewer than 12 months of data (for example 3 months), we would return a list of 3 objects to the frontend and in the case (2) there are at least 12 months of data, then we return the past 12 months of data, excluding the current month's data, regardless of whether the installs in any of those months is nonzero or zero, since we have already filtered the data so we know if months have zero installs then it is representative of what happened? @codemonkey800
